src/raftstore/store/peer.rs

```diff
@@ -1978,6 +1992,26 @@ impl Peer {
         Ok(())
     }
 
+    /// `ReadIndex` requests could be lost in network, so on followers commands could queue in
+    /// `pending_reads` forever. Sending a new `ReadIndex` periodically can resolve this.
+    pub(super) fn retry_pending_reads(&mut self) {
+        if self.is_leader() || self.pending_reads.reads.is_empty() || self.pre_read_index().is_err()
```

Comment:
Shall we clean the `pending_reads` with `StaleCommand` when the peer is not leader?